.github/workflows/AKSC_Deploy.yml

```diff
@@ -42,43 +47,60 @@ on:
         required: true
       USER_OBJECT_ID:
         required: false
+    outputs:
+      AKSNAME:
+        description: "The AKS Cluster name"
+        value: ${{ jobs.Deploy-AKSC.outputs.AKSNAME }}
 
 permissions:
   id-token: write
   contents: read
 
 jobs:
-  aksc_reusable_workflow:
+  Deploy-AKSC:
     runs-on: ubuntu-latest
+    environment: ${{ github.event.inputs.environment }}
+    outputs:
+      AKSNAME: ${{ steps.deployAks.outputs.AKSNAME }}
     steps:
       # Log into Azure
-    - uses: azure/login@v1
+    - uses: azure/login@v1.4.3
       with:
         client-id: ${{ secrets.AZURE_CLIENT_ID }}
         tenant-id: ${{ secrets.AZURE_TENANT_ID }}
         subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
 
       # Deploy Bicep file
     - name: Deploy Bicep
+      id: deployAks
       env:
-        templateParams: ${{ inputs.templateParams}}
+        templateParams: "${{ inputs.templateParams}}"
       run: |
-        if [ "${{ secrets.USER_OBJECT_ID }}" ]; then
+        if [ -n "${{ secrets.USER_OBJECT_ID }}" ]
+        then
           echo "Substitute _USER_OBJECT_ID_ placeholder value with real guid from secret"
           templateParams=${templateParams//=_USER_OBJECT_ID_/=${{ secrets.USER_OBJECT_ID }}}
+        else
+          echo "USER_OBJECT_ID not being leveraged in workflow"
         fi
-        az deployment group create -g ${{ inputs.rg }} --template-uri https://github.com/Azure/AKS-Construction/releases/download/${{ inputs.templateVersion }}/main.json --parameters ${{ inputs.templateParamFile}} automatedDeployment=true $templateParams
 
+        DEP=$(az deployment group create -g ${{ inputs.rg }} --template-uri https://github.com/Azure/AKS-Construction/releases/download/${{ inputs.templateVersion }}/main.json --parameters ${{ inputs.templateParamFile}} automatedDeployment=true $templateParams)
+        echo $DEP
```

Comment:
lots of echos here :) and you had to get jq in :).  I'd prefer to use --query if you just want one value, you dont need any followup processing then

Reply:
actually, we can also get acrName too, that would be nice

Reply:
have you got a example of $DEP, and I'll write the command

Reply:
I just like handling json as json not a string, besides JQ is on the runners 🎉

I don't have an example off hand.... but you can see the path here `.properties.outputs.aksClusterName.value`